script/citest.sh

```diff
@@ -74,10 +77,14 @@ export GITTEST_REMOTE_SSH_PUBKEY="$HOME/.ssh/id_rsa.pub"
 export GITTEST_REMOTE_SSH_PASSPHRASE=""
 ctest -V -R libgit2_clar-ssh || exit $?
 
-# Use the proxy we started at the beginning
-export GITTEST_REMOTE_PROXY_URL="localhost:8080"
-export GITTEST_REMOTE_PROXY_USER="foo"
-export GITTEST_REMOTE_PROXY_PASS="bar"
-ctest -V -R libgit2_clar-proxy_credentials || exit $?
+if [ -n "$JAVA_FOUND" ]; then
+	# Use the proxy we started at the beginning
+	export GITTEST_REMOTE_PROXY_URL="localhost:8080"
+	export GITTEST_REMOTE_PROXY_USER="foo"
+	export GITTEST_REMOTE_PROXY_PASS="bar"
+	ctest -V -R libgit2_clar-proxy_credentials || exit $?
+else
+	echo "Cannot find java in path, skipping proxy test"
+fi
```

Comment:
Why are you making this change?  Are there instances where travis doesn't have java installed?  Or is this for running locally?  If the latter, then let's make it some explicit opt-out.

Otherwise I fear that travis will change its images without us noticing, we'll stop running these tests silently, and then we'll break something without knowing.

Reply:
Yeah, this is slightly conflated with being able to run those scripts in a VM, hence this change (I didn't have `java` at first).

I understand the concern though, but on the other hand, changing that test made the proxy test run on the `osx` machines too. If I find a way to tell if the script is Travis run, I'll be able to make it hard error.